src/app/src/test/kotlin/org/imperial/mrc/comet/unit_tests/controllers/DataControllerTests.kt

```diff
@@ -0,0 +1,57 @@
+package org.imperial.mrc.comet.unit_tests.controllers
+
+import com.fasterxml.jackson.databind.node.ObjectNode
+import org.assertj.core.api.Assertions.assertThat
+import org.imperial.mrc.comet.controllers.DataController
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.slf4j.Logger
+import java.io.File
+import java.net.URL
+
+class DataControllerTests {
+    private val TMP_DIR = "tmp"
+
+    @BeforeEach
+    fun makeTmpDir() {
+        File(TMP_DIR).mkdir()
+    }
+
+    @AfterEach
+    fun cleanup() {
+        File(TMP_DIR).deleteRecursively()
+    }
+
+    private fun createTestResource(fileName: String, contents: String): URL {
+        val fullPath = "$TMP_DIR/fileName"
```

Comment:
```suggestion
        val fullPath = "$TMP_DIR/$fileName"
```
Or you could do something like `private val TMP_DIR = File("tmp")` above and then just do `val file = File(tmp, fileName)`, which might also be less OS dependent.